staging/src/k8s.io/dynamic-resource-allocation/resourceclaim/resourceclaim.go

```diff
@@ -114,25 +116,26 @@ func (l *Lookup) Name(pod *v1.Pod, podClaim *v1.PodResourceClaim) (name *string,
 			}
 			if info.Major == "" {
 				// Fake client...
-				l.usePodStatus = ptr.To(true)
+				usePodStatus = ptr.To(true)
 			} else {
 				switch strings.Compare(info.Major, "1") {
 				case -1:
 					// Huh?
-					l.usePodStatus = ptr.To(false)
+					usePodStatus = ptr.To(false)
 				case 0:
 					// info.Minor may have a suffix which makes it larger than 28.
 					// We don't care about pre-releases here.
-					l.usePodStatus = ptr.To(strings.Compare("28", info.Minor) <= 0)
+					usePodStatus = ptr.To(strings.Compare("28", info.Minor) <= 0)
 				case 1:
 					// Kubernetes 2? Yeah!
-					l.usePodStatus = ptr.To(true)
+					usePodStatus = ptr.To(true)
 				}
 			}
 		}
+		l.usePodStatus.Store(usePodStatus)
```

Comment:
usePodStatus can still be nil unchanged at this point as far as I can see. Would it make sense to only store it if it's changed?

Reply:
Which code path leaves `usePodStatus` unset? All of them should set it. `strings.Compare` only returns three different values.

Reply:
if env variable is empty and `l.client` is nil, usePodStatus will stay nil as far as I can see. Am I missing something?

Reply:
Well, if there is a chance that this can happen, then the code will panic on the line 138 below, trying to dereference nil pointer.

Reply:
That already was the case before. But you are right, this could happen as `NewNameLookup` is documented as "nil is allowed".

I've added the missing `else`.

Reply:
/lgtm